app/controllers/application_controller.rb

```diff
@@ -0,0 +1,19 @@
+class ApplicationController < ActionController::Base
+  # Prevent CSRF attacks by raising an exception.
+  # For APIs, you may want to use :null_session instead.
+  protect_from_forgery with: :exception
+
+  helper_method :current_user, :logged_in?
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id])
+  end
+
+  def logged_in?
+    !!current_user
+  end
+
+  def ensure_logged_in
```

Comment:
Think about making this a before_action on create, destroy, update, and then use skip_before_action in any controller you don't need to protect.